include/simdjson/generic/ondemand/json_iterator.h

```diff
@@ -89,7 +89,7 @@ class json_iterator {
   /**
    * Advance the current token.
    */
-  simdjson_really_inline const uint8_t *advance() noexcept;
+  simdjson_really_inline const uint8_t *return_current_and_advance() noexcept;
```

Comment:
To me...

```C++
  /**
    * Advance the current token.
    */
   simdjson_really_inline const uint8_t *advance() noexcept;
```

sounds like code that would advance and return the new data. Let us be super explicit that it is not what it does. It returns the current pointer and then advance the pointer (without accessing the new data yet).

Reply:
While advance() sounds different from next() to me, obviously it doesn't to everyone, so this is fine :)

I believe I picked this return_current_and_advance method (i++) over a simple next() (++i) mainly because I read there's an addressing mode that can get you the value and post-increment all in one go. It also makes it a little quicker to peek at the next token or get the length of the current token (since getting the token length involves subtracting *next from *current).

Anyway, this is good :) More explicit is better.